flwUpdate/src/main/java/org/motechproject/nms/flwUpdate/service/impl/FrontLineWorkerImportServiceImpl.java

```diff
@@ -94,12 +105,91 @@ public void importFrontLineWorker(Map<String, Object> record, State state) throw
             //It updated_date_nic from mcts is not null,then it's not a new record. Compare it with the record from database and update
             if (mctsUpdatedDateNic != null && (flw.getUpdatedDateNic() == null || mctsUpdatedDateNic.isAfter(flw.getUpdatedDateNic()) || mctsUpdatedDateNic.isEqual(flw.getUpdatedDateNic()))) {
                 frontLineWorkerService.update(updateFlw(flw, record, location));
+                Long oldMsisdn = flw.getContactNumber();
+                Long newMsisdn = (Long) record.get(FlwConstants.CONTACT_NO);
```

Comment:
Will add a check there. We don't have a confirmation but it's a mandatory field in mcts and the same is observed in RCH responses as well.